build/shade/_k-standard-goals.shade

```
@@ -151,24 +150,11 @@ default SAMPLES_PROJECT_GLOB = "samples/*/*.csproj"
               }
           }
       });
-
-      projectGlobs.AddRange(srcProjects);
-    }
-    if (!BuildSrcOnly && Directory.Exists("test"))
-    {
-        projectGlobs.AddRange(Files.Include(TEST_PROJECT_GLOB).ToList());
-    }
-    if (!BuildSrcOnly && Directory.Exists("samples") && Directory.EnumerateFiles("samples", "*.csproj", SearchOption.AllDirectories).Any())
-    {
-        projectGlobs.AddRange(Files.Include(SAMPLES_PROJECT_GLOB).ToList());
     }
 
-    if (projectGlobs.Any())
+    foreach (var sln in Files.Include(Path.Combine(BASE_DIR, "*.sln")))
```

Could we just build `$repoName.sln` - seems like a convention all our repos follow. We could throw an error if there happen to be more than one sln.

---

Yeah, that could be fine. I know off the top of my head that Diagnostics has DiagnosticPages.sln. Are there others?

---

HttpSysServer still has WebListener.sln though I think WebListener as a repo name will redirect properly

---

We also need to change that rename in aspnet/Universe's list of repos.

---

> We could throw an error if there happen to be more than one sln.

Meant to say we could throw an error to say we couldn't find the sln if we don't find it.

---

We could also just invoke "dotnet build" and let the CLI find the .sln file to use. If there are more than one files matching `*.*proj` or `*.sln` it will fail on its own.

---

| Repo | Solutions |
|------|-----------|
|Antiforgery|Antiforgery.sln|
|AzureIntegration|AzureIntegration.sln|
|BasicMiddleware|BasicMiddleware.sln|
|BrowserLink|BrowserLink.sln|
|Caching|Caching.sln|
|Common|Common.sln|
|Configuration|Configuration.sln|
|CORS|CORS.sln|
|DataProtection|DataProtection.sln|
|DependencyInjection|DependencyInjection.sln|
|Diagnostics|DiagnosticsPages.sln|
|DotNetTools|DotNetTools.sln|
|EntityFramework|EntityFramework.sln|
|EntityFramework.Tools|EntityFramework.Tools.sln|
|Entropy|Entropy.sln|
|EventNotification|EventNotification.sln|
|FileSystem|FileSystem.sln|
|Hosting|Hosting.sln|
|HtmlAbstractions|HtmlAbstractions.sln|
|HttpAbstractions|HttpAbstractions.sln|
|HttpSysServer|WebListener.sln|
|Identity|Identity.sln|
|IISIntegration|IISIntegration.sln|
|JsonPatch|JsonPatch.sln|
|KestrelHttpServer|KestrelHttpServer.sln|
|Localization|Localization.sln|
|Logging|Logging.sln|
|MetaPackages|MetaPackages.sln|
|Microsoft.Data.Sqlite|Microsoft.Data.Sqlite.sln|
|MusicStore|MusicStore.sln|
|Mvc|Mvc.NoFun.sln Mvc.sln|
|MvcPrecompilation|RazorViewCompilation.sln|
|Options|Options.sln|
|PlatformAbstractions|PlatformAbstractions.sln|
|Proxy|Proxy.sln|
|Razor|Razor.sln|
|ResponseCaching|ResponseCaching.sln|
|Routing|Routing.sln|
|Scaffolding|Scaffolding.sln|
|Security|Security.sln|
|ServerTests|ServerTests.sln|
|Session|Session.sln|
|SignalR|SignalR.sln|
|StaticFiles|StaticFiles.sln|
|Testing|Testing.sln|
|WebSockets|WebSockets.sln|

---

MvcPrecompilation, Diagnostics and HttpSysServer are the only outliers here. Should be easy to fix them.

---

The only problematic thing is...how do you determine "$repoName"? We can't use directory name as it's not guaranteed....

---

That's true. I guess we should get Mvc to move the second sln (the NoFun one) out of the repo root as part of the migration. We'll switch to calling `dotnet build` and have it fail in the non-common cases.

FYI @NTaylorMullen

---

Bummer, was working around some issues with having multiple sln's in Mvc's [migrated makefile](https://github.com/aspnet/Mvc/blob/dbcb56f09b97d6ac7c05ace96bd7bf6d99331c57/makefile.shade#L8). Could we take the approach of having a parameter that we can set to override the default sln name?. It'd be a shame to hide sln's that we actually use from users who clone our repos.

---

Instead of a parameter, we could build a well known file (`build.proj`?) if it's present. We can't run `dotnet build` regardless.

---

I think this problem can be solved by the work Pranav is doing on https://github.com/aspnet/KoreBuild/pull/158. The default Korebuild build step could include:

```xml
<SolutionsToBuild Include="$(RepoRoot)\*.sln" />
```
MVC and other repos could customize by adding exclusions to the `build/repo.targets` file:

```xml
<SolutionsToBuild Remove="$(RepoRoot)\Mvc.NoFun.sln" />
```